visual-tests/imports/helpers.js

```diff
@@ -0,0 +1,67 @@
+export const helpers = {
+  defaultHOTWrapper: 'js',
+
+  screenshotsDirectory: './tests/screenshots',
+  screenshotsExtension: 'png',
+
+  selectors: {
+    mainTable: 'div.handsontable.htRowHeaders.htColumnHeaders',
+    mainTableBody: '> .ht_master.handsontable table tbody',
+    mainTableHead: '> .ht_clone_top.handsontable table thead',
+    mainTableFirstColumn: '> .ht_clone_inline_start.ht_clone_left.handsontable table tbody',
+    dropdownMenu: '.htMenu.htDropdownMenu.handsontable',
+  },
+
+  cssFiles: {
+    cookieInfo: 'remove-cookie-info.css',
+    changeCellsBackground: 'change-cells-background.css',
+    changeCellTextColor: 'change-cell-text-color.css',
+    dynamicDataFreeze: 'dynamic-data-freeze.css'
+  },
+
+  expectedPageTitle: /Handsontable for .* example/,
+
+  cssPath(file) {
+    return `./tests-css/${file}`;
+  },
+
+  init(workerInfo, process) {
```

Comment:
```suggestion
  init(workerInfo) {
```
The `process` variable is already available globally in the Node. I don't see reasons to pass it in tests. Or am I missing something?

Reply:
>     * [#10165 (comment)](https://github.com/handsontable/handsontable/pull/10165#discussion_r1058142185)
>       This comment is still applicable, for example:
>       
>       * `mainTable: 'div.handsontable.htRowHeaders.htColumnHeaders'` can still easily target more than one element (as described in the original comment).
>         https://jsfiddle.net/js_ziggle/e0dgwap8/ - start editing a cell with handsontable-based editor and run `querySelectorAll` with the above selector.

hey, this thing required little more changes, not only in visual testing, so was done just after your comment, but in different PR - it is available here: https://github.com/handsontable/handsontable/pull/10193

Was pretty sure that I told you about it, but now I see that link is missing.

>       * `mainTableFirstColumn: '> .ht_clone_inline_start.ht_clone_left.handsontable table tbody'` in the vast majority of instances doesn't target "first column" - it targets the "left overlay table" (as described in the original comment)

About last point - as far I remember we wasn't sure how to make it flexible, but actually I'm not sure if in our case it is necessary to change this part, cause there is a chance that described case will never happen ( as far as I know we will be testing always this and the same configuration of HoT ).

>     * [#10165 (comment)](https://github.com/handsontable/handsontable/pull/10165#discussion_r1056297438)
>       You didn't respond to this comment in any way, could you clarify if you think it's a good or a bad idea?

My bad, looks like I misundertood this point and thought that it is automatically outdated after improving another thing. Will be improved this week.

>     * [#10165 (comment)](https://github.com/handsontable/handsontable/pull/10165#discussion_r1057565222)
>       If it's a leftover, maybe we should change it to the actual license?

Sure - should I just copy it from other file?

>       **This one is the most important issue** - if I'm seeing this correctly, the script is still not testing the `next` build with the _local builds_. It just installs whatever's in the examples' `package.json` files.

Well, this is the point I had mentioned on last meeting that still can be broken and requiring support - and looks like it is. I do not know process of building examples well and looks like my changes was not enough, but I asked about some introducing and then everything will be clear.

> @maciejbo-handsontable I haven't looked through all the changes since my last review yet, but I noticed these things:
> 
>     * [#10165 (comment)](https://github.com/handsontable/handsontable/pull/10165#discussion_r1058142185)
>       This comment is still applicable, for example:
>       
>       * `mainTable: 'div.handsontable.htRowHeaders.htColumnHeaders'` can still easily target more than one element (as described in the original comment).
>         https://jsfiddle.net/js_ziggle/e0dgwap8/ - start editing a cell with handsontable-based editor and run `querySelectorAll` with the above selector.
>       * `mainTableFirstColumn: '> .ht_clone_inline_start.ht_clone_left.handsontable table tbody'` in the vast majority of instances doesn't target "first column" - it targets the "left overlay table" (as described in the original comment)
> 
>     * [#10165 (comment)](https://github.com/handsontable/handsontable/pull/10165#discussion_r1056297438)
>       You didn't respond to this comment in any way, could you clarify if you think it's a good or a bad idea?
> 
>     * [#10165 (comment)](https://github.com/handsontable/handsontable/pull/10165#discussion_r1057565222)
>       If it's a leftover, maybe we should change it to the actual license?
> 
>     * [#10165 (review)](https://github.com/handsontable/handsontable/pull/10165#pullrequestreview-1228873761)
>       **This one is the most important issue** - if I'm seeing this correctly, the script is still not testing the `next` build with the _local builds_. It just installs whatever's in the examples' `package.json` files.
> 
> 
> Additionally: I tried running the `npm run test` script locally after changing the `baseBranch` to the branch I'm on, but I get the following error:
> 
> ```
> > handsontable-visual-tests@0.0.0 test
> > node ./scripts/visual-tests-launcher.mjs build
> 
> Unable to find image 'vtests:latest' locally
> docker: Error response from daemon: pull access denied for vtests, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
> See 'docker run --help'.
> /Users/jansiegel/workspace/cr-handsontable/visual-tests/node_modules/execa/lib/error.js:60
> 		error = new Error(message);
> 		        ^
> 
> Error: Command failed with exit code 125: docker run --name vtests-container --env CURRENT_BRANCH=feature/dev-issue-572 -v /Users/jansiegel/workspace/cr-handsontable/visual-tests/../:/vtests/ -w /vtests/visual-tests --rm vtests
>     at makeError (/Users/jansiegel/workspace/cr-handsontable/visual-tests/node_modules/execa/lib/error.js:60:11)
>     at handlePromise (/Users/jansiegel/workspace/cr-handsontable/visual-tests/node_modules/execa/index.js:118:26)
>     at processTicksAndRejections (node:internal/process/task_queues:96:5)
>     at async file:///Users/jansiegel/workspace/cr-handsontable/visual-tests/scripts/visual-tests-launcher.mjs:32:5 {
>   shortMessage: 'Command failed with exit code 125: docker run --name vtests-container --env CURRENT_BRANCH=feature/dev-issue-572 -v /Users/jansiegel/workspace/cr-handsontable/visual-tests/../:/vtests/ -w /vtests/visual-tests --rm vtests',
>   command: 'docker run --name vtests-container --env CURRENT_BRANCH=feature/dev-issue-572 -v /Users/jansiegel/workspace/cr-handsontable/visual-tests/../:/vtests/ -w /vtests/visual-tests --rm vtests',
>   escapedCommand: 'docker run --name vtests-container --env "CURRENT_BRANCH=feature/dev-issue-572" -v "/Users/jansiegel/workspace/cr-handsontable/visual-tests/../:/vtests/" -w "/vtests/visual-tests" --rm vtests',
>   exitCode: 125,
>   signal: undefined,
>   signalDescription: undefined,
>   stdout: undefined,
>   stderr: undefined,
>   failed: true,
>   timedOut: false,
>   isCanceled: false,
>   killed: false
> }
> ```
> 
> Am I doing something wrong?

Reply:
probably it's kind of leftover. Modified.